include/seqan3/alphabet/composite/alphabet_variant.hpp

```diff
@@ -28,6 +29,22 @@
 namespace seqan3::detail
 {
 
+//!\brief Helper to determine whether a type is a std::span.
+template <typename t, size_t extent = std::dynamic_extent>
+struct is_span
+{
+    //!\brief `true` if `t` is a std::span, otherwise `false`.
+    static constexpr bool value = false;
+};
+
+//!\cond
+template <typename t, size_t extent>
+struct is_span<std::span<t, extent>>
+{
+    static constexpr bool value = true;
+};
+//!\endcond
+
```

Comment:
This can (I believe) be reduced and clarified by using `std::{true,false}_type`.
```suggestion
template <typename t, size_t extent = std::dynamic_extent>
struct is_span : std::false_type {};

//!\cond
template <typename t, size_t extent>
struct is_span<std::span<t, extent>> : std::true_type {};
//!\endcond

```